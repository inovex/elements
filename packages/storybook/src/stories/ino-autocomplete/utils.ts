import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
