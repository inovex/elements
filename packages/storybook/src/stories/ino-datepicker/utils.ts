import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
