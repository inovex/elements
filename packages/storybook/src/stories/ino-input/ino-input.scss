.story-ino-input {
  ino-input {
    margin-top: 15px;
    margin-bottom: 15px;
  }
}
