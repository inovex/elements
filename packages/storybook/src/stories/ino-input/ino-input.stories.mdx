import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-input/readme.md";

<Meta title="Input/ino-input/Documentation" />

<Description markdown={readme} />
