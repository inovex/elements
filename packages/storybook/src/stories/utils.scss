/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
