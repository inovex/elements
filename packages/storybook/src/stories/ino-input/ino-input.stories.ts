import { useEffect } from '@storybook/client-api';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import './ino-input.scss';

export default {
  title: 'Input/ino-input',
  component: 'ino-input',
  decorators: [
    (story) => defaultDecorator(story, 'story-ino-input'),
    (story) => {
      useEffect(() => {
        const eventHandler = (e) => e.target.setAttribute('value', e.detail);
        const inputs = document.querySelectorAll('ino-input');
        inputs.forEach((i) => i.addEventListener('valueChange', eventHandler));
        return () =>
          inputs.forEach((i) =>
            i.removeEventListener('valueChange', eventHandler)
          );
      });
      return story();
    },
  ],
  parameters: {
    actions: {
      handles: [
        'iconClick .customizable-input',
        'valueChange .customizable-input',
      ],
    },
  },
};

export const Playground = (args) => html`
  <ino-input
    class="customizable-input"
    id="customizable-input"
    autocomplete="${args.autcomplete}"
    autofocus="${args.autoFocus}"
    data-list="${args.dataList}"
    decimal-places="${args.decimalPlaces}"
    disabled="${args.disabled}"
    error="${args.error}"
    helper="${args.helper}"
    helper-character-counter="${args.helperCharacterCounter}"
    helper-persistent="${args.helperPersistent}"
    helper-validation="${args.helperValidation}"
    label="${args.label}"
    min="${args.min}"
    max="${args.max}"
    maxlength="${args.maxlength}"
    name="${args.name}"
    outline="${args.outline}"
    pattern="${args.pattern}"
    placeholder="${args.placeholder}"
    readonly="${args.readonly}"
    required="${args.required}"
    show-label-hint="${args.showLabelHint}"
    size=${args.size}
    step="${args.step}"
    thousands-separator="${args.thousandsSeparator}"
    type="${args.type}"
    unit="${args.unit}"
    value="${args.value}"
  >
  </ino-input>
`;
Playground.args = {
  autocomplete: false,
  autoFocus: false,
  dataList: '',
  decimalPlaces: '',
  disabled: false,
  error: false,
  helper: 'Helper message',
  helperCharacterCounter: false,
  helperPersistent: false,
  helperValidation: false,
  label: 'Input label',
  min: '',
  max: '',
  maxlength: 100,
  outline: false,
  name: '',
  pattern: '*',
  placeholder: '',
  readonly: false,
  required: false,
  showLabelHint: false,
  size: '',
  step: 5,
  thousandsSeparator: false,
  type: 'text',
  unit: '',
  value: '',
};
Playground.argTypes = {
  type: {
    control: {
      type: 'select',
    },
    options: ['text', 'number', 'password', 'email'],
  },
};

export const Types = () => html`
  <ino-input placeholder="type = text"></ino-input>
  <ino-input placeholder="type = email" type="email"></ino-input>
  <ino-input type="number" placeholder="type = number"></ino-input>
  <ino-input
    type="number"
    placeholder="type = number with steps=5"
    step="5"
  ></ino-input>
  <ino-input
    type="number"
    placeholder="type = number with step='any' (decimal numbers)"
    step="any"
  ></ino-input>
  <ino-input type="password" placeholder="type = password"></ino-input>
`;

export const States = () => html`
  <ino-input placeholder="Disabled" disabled></ino-input>
  <ino-input label="Optional" show-label-hint></ino-input>
  <ino-input label="Required" required show-label-hint></ino-input>

  <ino-input placeholder="Outline disabled" outline disabled></ino-input>
  <ino-input label="Outline optional" outline show-label-hint></ino-input>
  <ino-input
    label="Outline required"
    outline
    show-label-hint
    required
  ></ino-input>
`;

export const Labels = () => html`
  <ino-input label="Floating label"></ino-input>
  <ino-input label="Floating label" value="With value"></ino-input>

  <ino-input label="Outlined floating label" outline></ino-input>
  <ino-input
    label="Outlined floating label"
    value="With value"
    outline
  ></ino-input>
`;

export const HelperTexts = () => html`
  <ino-input
    placeholder="Helper text on focus (default)"
    helper="Helper text"
  ></ino-input>
  <ino-input
    placeholder="Persistent helper text"
    helper="Persistent helper text"
    helper-persistent
  ></ino-input>
  <ino-input
    required
    placeholder="Validation message for required input"
    helper="This field is required"
    helper-validation
  ></ino-input>
  <ino-input
    value="Length of 12"
    helper-character-counter
    maxlength="25"
  ></ino-input>
`;

export const Outlined = () => html`
  <ino-input placeholder="Outlined" outline></ino-input>
`;

export const Icons = () => html`
  <ino-input icon-leading label="Leading icon">
    <ino-icon slot="icon-leading" icon="search"></ino-icon>
  </ino-input>
  <ino-input icon-trailing label="Trailing icon">
    <ino-icon slot="icon-trailing" icon="search"></ino-icon>
  </ino-input>
  <ino-input icon-leading icon-trailing label="Leading and Trailing icon">
   <ino-icon slot="icon-leading" icon="search"></ino-icon>
   <ino-icon slot="icon-trailing" icon="add"></ino-icon>
  </ino-input>
  <ino-input icon-trailing label="Clickable Icon">
    <ino-icon clickable slot="icon-trailing" icon="search"></ino-icon>
  </ino-input>
 </div>
`;

export const DataList = () => html`
  <ino-input label="Datalist" data-list="languages">
    <datalist id="languages">
      <option>HTML</option>
      <option>Java</option>
      <option>Perl</option>
      <option>PHP</option>
      <option>Ruby on Rails</option>
    </datalist>
  </ino-input>
`;

export const NumberFormats = () => html`
  <ino-input decimal-places="2" placeholder="With 2 decimal places"></ino-input>
  <ino-input
    thousands-separator
    placeholder="With thousands separator"
  ></ino-input>
`;

export const Metadata = () => html`
  <ino-input
    value="2,00"
    decimal-places="2"
    thousands-separator
    unit="â‚¬"
    label="Euro Input"
  ></ino-input>
  <ino-input value="2" type="number" unit="h" label="Hours input"></ino-input>
`;
