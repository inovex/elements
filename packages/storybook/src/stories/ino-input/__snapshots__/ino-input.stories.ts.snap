// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-input DataList smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input label="Datalist"
               data-list="languages"
               class="hydrated"
               tabindex="-1"
    >
      <datalist id="languages">
        <option>
          HTML
        </option>
        <option>
          Java
        </option>
        <option>
          Perl
        </option>
        <option>
          PHP
        </option>
        <option>
          Ruby on Rails
        </option>
      </datalist>
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Datalist
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
               list="languages"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input HelperTexts smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input placeholder="Helper text on focus (default)"
               helper="Helper text"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="Helper text on focus (default)"
               type="text"
               aria-controls="input-helper-text__17"
               aria-describedby="input-helper-text__17"
        >
      </label>
      <div class="mdc-text-field-helper-line">
        <div class="mdc-text-field-helper-text"
             id="input-helper-text__17"
             aria-hidden="true"
        >
          Helper text
        </div>
      </div>
    </ino-input>
    <ino-input placeholder="Persistent helper text"
               helper="Persistent helper text"
               helper-persistent
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="Persistent helper text"
               type="text"
               aria-controls="input-helper-text__18"
               aria-describedby="input-helper-text__18"
        >
      </label>
      <div class="mdc-text-field-helper-line">
        <div class="mdc-text-field-helper-text mdc-text-field-helper-text--persistent"
             id="input-helper-text__18"
             aria-hidden="true"
        >
          Persistent helper text
        </div>
      </div>
    </ino-input>
    <ino-input required
               placeholder="Validation message for required input"
               helper="This field is required"
               helper-validation
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="ino-label--required hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="Validation message for required input"
               required
               type="text"
               aria-controls="input-helper-text__19"
        >
      </label>
      <div class="mdc-text-field-helper-line">
        <div class="mdc-text-field-helper-text mdc-text-field-helper-text--validation-msg"
             id="input-helper-text__19"
             aria-hidden="true"
        >
          This field is required
        </div>
      </div>
    </ino-input>
    <ino-input value="Length of 12"
               helper-character-counter
               maxlength="25"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               maxlength="25"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
        <div class="mdc-text-field-character-counter">
          12 / 25
        </div>
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input Icons smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input icon-leading
               label="Leading icon"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--with-leading-icon mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Leading icon
          </label>
        </ino-label>
        <span class="mdc-text-field__icon mdc-text-field__icon--leading">
          <ino-icon slot="icon-leading"
                    icon="search"
                    class="hydrated ino-icon--color-scheme-primary"
          >
          </ino-icon>
        </span>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input icon-trailing
               label="Trailing icon"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--with-trailing-icon mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Trailing icon
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
        <span class="mdc-text-field__icon mdc-text-field__icon--trailing">
          <ino-icon slot="icon-trailing"
                    icon="search"
                    class="hydrated ino-icon--color-scheme-primary"
          >
          </ino-icon>
        </span>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input icon-leading
               icon-trailing
               label="Leading and Trailing icon"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--with-leading-icon mdc-text-field--with-trailing-icon mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Leading and Trailing icon
          </label>
        </ino-label>
        <span class="mdc-text-field__icon mdc-text-field__icon--leading">
          <ino-icon slot="icon-leading"
                    icon="search"
                    class="hydrated ino-icon--color-scheme-primary"
          >
          </ino-icon>
        </span>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
        <span class="mdc-text-field__icon mdc-text-field__icon--trailing">
          <ino-icon slot="icon-trailing"
                    icon="add"
                    class="hydrated ino-icon--color-scheme-primary"
          >
          </ino-icon>
        </span>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input icon-trailing
               label="Clickable Icon"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--with-trailing-icon mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Clickable Icon
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
        <span class="mdc-text-field__icon mdc-text-field__icon--trailing">
          <ino-icon clickable
                    slot="icon-trailing"
                    icon="search"
                    class="hydrated ino-icon--color-scheme-primary"
          >
          </ino-icon>
        </span>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input Labels smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input label="Floating label"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Floating label
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Floating label"
               value="With value"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--label-floating mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label mdc-floating-label--float-above">
            Floating label
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Outlined floating label"
               outline
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--outlined">
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Outlined floating label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Outlined floating label"
               value="With value"
               outline
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--outlined mdc-text-field--label-floating">
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch"
                 style="width: 127.25px;"
            >
              <label class="mdc-floating-label mdc-floating-label--float-above"
                     style
              >
                Outlined floating label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input Metadata smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input value="2"
               type="number"
               unit="h"
               label="Hours input"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--with-trailing-icon mdc-text-field--label-floating mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label mdc-floating-label--float-above">
            Hours input
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="number"
        >
        <span class="mdc-text-field__affix mdc-text-field__affix--suffix">
          h
        </span>
        <div class="arrow-container">
          <ino-icon class="ino-num-arrows up ino-icon--color-scheme-primary hydrated">
          </ino-icon>
          <ino-icon class="ino-num-arrows down ino-icon--color-scheme-primary hydrated">
          </ino-icon>
        </div>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input NumberFormats smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input type="number"
               decimal-places="2"
               placeholder="With 2 decimal places"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="With 2 decimal places"
               type="number"
        >
        <div class="arrow-container">
          <ino-icon class="ino-num-arrows up ino-icon--color-scheme-primary hydrated">
          </ino-icon>
          <ino-icon class="ino-num-arrows down ino-icon--color-scheme-primary hydrated">
          </ino-icon>
        </div>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input type="number"
               step="1"
               thousands-separator
               placeholder="With thousands separator"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="With thousands separator"
               type="number"
        >
        <div class="arrow-container">
          <ino-icon class="ino-num-arrows up ino-icon--color-scheme-primary hydrated">
          </ino-icon>
          <ino-icon class="ino-num-arrows down ino-icon--color-scheme-primary hydrated">
          </ino-icon>
        </div>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input class="customizable-input hydrated"
               id="customizable-input"
               autocomplete="off"
               autofocus="false"
               data-list
               disabled="false"
               error="false"
               helper="Helper message"
               helper-character-counter="false"
               helper-persistent="false"
               helper-validation="false"
               label="Input label"
               min
               max
               maxlength="100"
               name
               outline="false"
               pattern="*"
               placeholder
               required="false"
               show-label-hint="false"
               size="0"
               step="5"
               type="text"
               unit
               value
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Input label
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               autocomplete="off"
               min
               max
               maxlength="100"
               step="5"
               name
               pattern="*"
               placeholder
               type="text"
               aria-controls="input-helper-text__0"
               aria-describedby="input-helper-text__0"
               list
        >
      </label>
      <div class="mdc-text-field-helper-line">
        <div class="mdc-text-field-helper-text"
             id="input-helper-text__0"
             aria-hidden="true"
        >
          Helper message
        </div>
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input States smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input placeholder="Disabled"
               disabled
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--disabled mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="ino-label--disabled hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               disabled
               step="1"
               placeholder="Disabled"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Optional"
               show-label-hint
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-ripple-upgraded">
        <ino-label show-hint
                   class="ino-label--show-hint hydrated"
        >
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label">
            Optional
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Required"
               required
               show-label-hint
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-ripple-upgraded">
        <ino-label show-hint
                   class="ino-label--show-hint ino-label--required hydrated"
        >
          <div class="mdc-line-ripple">
          </div>
          <label class="mdc-floating-label mdc-floating-label--required">
            Required
          </label>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               required
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input placeholder="Outline disabled"
               outline
               disabled
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--disabled mdc-text-field--outlined mdc-text-field--no-label">
        <ino-label class="ino-label--outlined ino-label--disabled hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--no-label">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               disabled
               step="1"
               placeholder="Outline disabled"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Outline optional"
               outline
               show-label-hint
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--outlined">
        <ino-label show-hint
                   class="ino-label--show-hint ino-label--outlined hydrated"
        >
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Outline optional
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input label="Outline required"
               outline
               show-label-hint
               required
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--outlined">
        <ino-label show-hint
                   class="ino-label--show-hint ino-label--outlined ino-label--required hydrated"
        >
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label mdc-floating-label--required"
                     style
              >
                Outline required
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               required
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;

exports[`Input/ino-input Types smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-input">
    <ino-input placeholder="type = text"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="type = text"
               type="text"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input placeholder="type = email"
               type="email"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="type = email"
               type="email"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input type="number"
               placeholder="type = number"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="type = number"
               type="number"
        >
        <div class="arrow-container">
          <ino-icon class="ino-num-arrows up hydrated ino-icon--color-scheme-primary">
          </ino-icon>
          <ino-icon class="ino-num-arrows down hydrated ino-icon--color-scheme-primary">
          </ino-icon>
        </div>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input type="number"
               placeholder="type = number with steps=5"
               step="5"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="5"
               placeholder="type = number with steps=5"
               type="number"
        >
        <div class="arrow-container">
          <ino-icon class="ino-num-arrows up hydrated ino-icon--color-scheme-primary">
          </ino-icon>
          <ino-icon class="ino-num-arrows down hydrated ino-icon--color-scheme-primary">
          </ino-icon>
        </div>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input type="number"
               placeholder="type = number with step='any' (decimal numbers)"
               step="any"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="any"
               placeholder="type = number with step='any' (decimal numbers)"
               type="number"
        >
        <div class="arrow-container">
          <ino-icon class="ino-num-arrows up hydrated ino-icon--color-scheme-primary">
          </ino-icon>
          <ino-icon class="ino-num-arrows down hydrated ino-icon--color-scheme-primary">
          </ino-icon>
        </div>
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
    <ino-input type="password"
               placeholder="type = password"
               class="hydrated"
               tabindex="-1"
    >
      <label class="ino-input__composer mdc-text-field mdc-text-field--filled mdc-text-field--box mdc-text-field--no-label mdc-ripple-upgraded">
        <ino-label class="hydrated">
          <div class="mdc-line-ripple">
          </div>
        </ino-label>
        <input class="mdc-text-field__input"
               step="1"
               placeholder="type = password"
               type="password"
        >
      </label>
      <div class="mdc-text-field-helper-line">
      </div>
    </ino-input>
  </div>
</div>
`;
